### PR TITLE
Provide Cadencefx bindings

### PR DESCRIPTION
**Pull request by ntlimited**
This adds the subpackage `cadencefx` to the client library, providing a set of Fx options that developers can use to easily build Fx applications with cadence workers.

### CODE REVIEW

File: cadencefx/mocks/Interface.go [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+
+import yarpc "go.uber.org/yarpc"
+
+// Interface is an autogenerated mock type for the Interface type
+type Interface struct {
+	mock.Mock
+}
+
+// DeprecateDomain provides a mock function with given fields: ctx, DeprecateRequest, opts
+func (_m *Interface) DeprecateDomain(ctx context.Context, DeprecateRequest *shared.DeprecateDomainRequest, opts ...yarpc.CallOption) error {
```
Comment (yiminc) [inline cadencefx/mocks/Interface.go]: We already have mocks defined under cadence/mocks package. Can we reuse that?

Comment (ntlimited) [reply cadencefx/mocks/Interface.go]: moved mocks

File: cadencefx/worker.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+// host specific task list specified by TaskList.
+//
+// A host specific task list is unique to the host the worker is running on, thus, scheduling a task to a host specific tasklist
+// forces the activity to be processed only by workers on the host. The main use case for this feature is with workflows that
+// require all subsequent activities be completed on the same node as its first activity. In other words, workflows that store
+// some or all of its working state on a node and therefore must have the entirety of its run stickied to a host.
+//
+// A Provider function using this struct could look like:
+//
+// func newHostSpecificTaskList() *HostSpecificTaskListConfig {
```
Comment (dmetzgar) [inline cadencefx/worker.go]: I think this needs to be indented so it's considered preformatted text by godoc

File: cadencefx/cadencefx.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+var Module = fx.Options(
+	ClientModule,
+	fx.Provide(NewWorkers),
+	fx.Invoke(RegisterWorkers),
+)
+
+// ClientModule provides constructors for:
+// 1. Cadence thrift client
+// 2. Cadence client.Client
+// 3. Cadence domain client
```
Comment (dmetzgar) [inline cadencefx/cadencefx.go]: It seems godoc still [doesn't support lists](https://github.com/golang/go/issues/7873). So I think this needs to be indented so it doesn't get treated like a single paragraph.

File: README.md [hunk 4]
```diff
@@ -140,59 +143,82 @@ var ClientName = "SimpleWorker"
...
 
-func startWorker(logger *zap.Logger, service workflowserviceclient.Interface) {
+func provideWorker(
+	lc fx.Lifecycle,
+	logger *zap.Logger,
+	service workflowserviceclient.Interface,
+) worker.Worker {
 	// TaskListName - identifies set of client workflows, activities and workers.
 	// it could be your group or client or application name.
 	workerOptions := worker.Options{
```
Comment (yiminc) [inline README.md]: how does user specify other worker options? Like MaxConcurrentDecisionTaskExecutionSize, or ActivityBackgroundContext?

File: cadencefx/cadencefx.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+//    client:
+//	  - domain: samples-domain
+//
+// In addition, yarpc needs to know about cadence thrift client:
+// yarpc:
+//   outbounds:
+//     cadence:
+//       service: cadence-frontend
+//       tchannel:
+//         peer: "127.0.0.1:7933"
```
Comment (yiminc) [inline cadencefx/cadencefx.go]: how to specify with muttley?

**Comment by mfateev** [comment]
Customer requested ability to disable Cadence through:
```
cadence:
  disable: true
```

**Pull request closed without merging**